runtime-sdk-macros/src/method_handler_derive.rs

```diff
@@ -232,6 +233,42 @@ pub fn derive_method_handler(impl_block: syn::ItemImpl) -> TokenStream {
         }
     };
 
+    let allowed_confidential_queries_impl = {
+        let handler_names: Vec<syn::Expr> = handlers
+            .iter()
+            .filter_map(|h| h.handler.as_ref())
+            .filter(|h| h.attrs.kind == HandlerKind::Query && h.attrs.allow_confidential)
+            .map(|h| h.attrs.rpc_name.clone())
+            .collect();
+        if handler_names.is_empty() {
+            quote! {}
+        } else {
+            quote! {
+                fn allowed_confidential_queries() -> Vec<&'static str> {
+                    vec![ #( #handler_names, )* ]
+                }
```

Comment:
```suggestion
                fn allowed_confidential_queries() -> &'static [&'static str] {
                    &[ #( #handler_names, )* ]
                }
```
to avoid an allocation. You could also do
```
                fn allowed_confidential_queries() -> [&'static str; #handler_count] {
                    [ #( #handler_names, )* ]
                }
```
if you want to operate over them and don't like the double-deref required by `Iterator` methods.

Or you could do neither. I just hate using the system allocator. It's so slow.

Reply:
Good point. But currently the problem is the tuple implementations need to merge the lists.

I guess we don't really need to get a list of everything, we just need an iterator over all or even just a check that returns true if the given method is in the set.